racket/collects/racket/contract/private/orc.rkt

```diff
@@ -14,65 +15,75 @@
                      [_flat-murec-contract flat-murec-contract]))
 
 (define/subexpression-pos-prop/name or/c-name or/c
-  (case-lambda 
-    [() (make-none/c '(or/c))]
-    [(x) (coerce-contract 'or/c x)]
-    [raw-args
-     (define args (coerce-contracts 'or/c raw-args))
-     (define-values (ho-contracts flat-contracts)
-       (let loop ([ho-contracts '()]
-                  [flat-contracts '()]
-                  [args args])
-         (cond
-           [(null? args) (values (reverse ho-contracts) (reverse flat-contracts))]
-           [else 
-            (let ([arg (car args)])
-              (cond
-                [(flat-contract? arg)
-                 (loop ho-contracts (cons arg flat-contracts) (cdr args))]
-                [else
-                 (loop (cons arg ho-contracts) flat-contracts (cdr args))]))])))
-     (define pred (make-flat-predicate flat-contracts))
-     (define the-or/c
-       (cond
-         [(null? ho-contracts)
-          (cond
-            [(and (pair? flat-contracts)
-                  (pair? (cdr flat-contracts))
-                  (null? (cddr flat-contracts))
-                  (or (and (equal? false/c-contract (car flat-contracts))
-                           (equal? true/c-contract (cadr flat-contracts)))
-                      (and (equal? false/c-contract (cadr flat-contracts))
-                           (equal? true/c-contract (car flat-contracts)))))
-             (coerce-contract 'or/c boolean?)]
-            [else
-             (make-flat-or/c pred flat-contracts)])]
-         [(null? (cdr ho-contracts))
-          (define name (apply build-compound-type-name 'or/c args))
-          (if (chaperone-contract? (car ho-contracts))
-              (make-chaperone-single-or/c name pred flat-contracts (car ho-contracts))
-              (make-impersonator-single-or/c name pred flat-contracts (car ho-contracts)))]
-         [else
-          (define name (apply build-compound-type-name 'or/c args))
-          (if (andmap chaperone-contract? ho-contracts)
-              (make-chaperone-multi-or/c name flat-contracts ho-contracts)
-              (make-impersonator-multi-or/c name flat-contracts ho-contracts))]))
-     (cond
-       [(ormap prop:any/c? args) (named-any/c (contract-name the-or/c))]
-       [else the-or/c])]))
+  (let ([none? (make-none/c '(or/c))])
```

Comment:
I don't think adding this let is a good idea. We are already using `none?` as somethign elsewhere here and it is a different thing than this. Also, the let change the indentation, making the diff more painful to work through; in this case we can do this without an indentation change by using `define`.